static/js/components/SearchResult.js

```diff
@@ -83,22 +89,42 @@ const ProfileSearchResult = ({ result }: ProfileProps) => {
   )
 }
 
-type CourseProps = {
-  result: CourseResult,
+type LearningResourceProps = {
+  result: CourseResult | BootcampResult,
   toggleFacet?: Function,
-  setShowCourseDrawer?: ({ courseId: string }) => void
+  setShowResourceDrawer?: ({ objectId: string, objectType: string }) => void
 }
+
 const CourseSearchResult = ({
   result,
   toggleFacet,
-  setShowCourseDrawer
-}: CourseProps) => {
+  setShowResourceDrawer
+}: LearningResourceProps) => {
+  // $FlowFixMe: this should only be used for courses
   const course = searchResultToCourse(result)
   return (
     <CourseCard
-      course={course}
+      object={course}
+      objectType="course"
       toggleFacet={toggleFacet}
-      setShowCourseDrawer={setShowCourseDrawer}
+      setShowResourceDrawer={setShowResourceDrawer}
+    />
+  )
+}
+
+const BootcampSearchResult = ({
+  result,
+  toggleFacet,
+  setShowResourceDrawer
+}: LearningResourceProps) => {
+  // $FlowFixMe: this should only be used for bootcamps
+  const bootcamp = searchResultToBootcamp(result)
```

Comment:
one other approach I was thinking that we could take here is to, instead of having a separate `searchResultToBootcamp` we could modify the existing `searchResultToCourse` to be `searchResultToLearningResource` to sort of follow-along with how the UI is organized. That function could set `platform` to be `bootcamps` when the `object_type` is `bootcamp`, and then that value could be used throughout the UI to control whether we show bootcamp-specific UI or not.

Does that make sense? if we did it that way we'd avoid the need for a separate "bootcamp" search result, we wouldn't need to pass down the "objectType" parameter, etc because anywhere we need to figure out how to render the thing we could just look at the `platform` value.

Reply:
I can make a `searchResultToLearningResource` that then calls the existing `searchResultToCourse` or `searchResultToBootcamp` function depending on the `object_type`.  The `Bootcamp` type does not include `platform` or `level` or `semester`), and not sure if `platform` makes sense anyway for the upcoming `LearningPath` results, which will be very different than bootcamps and courses.

Reply:
yeah, I was thinking instead of even having separate `Course` and `Bootcamp` types why not have a single `LearningResource`? the two `Course` and `Bootcamp` types you have now have the majority of their fields in common already.

Reply:
Don't I need different types since they have different fields, even if currently it's just a few, otherwise flow will start complaining? I could have a `LearningResource` that both the `Course` and `Bootcamp` types inherit from with common fields, but once I introduce `Program` and `LearningPath` the common fields will go down substantially.

Reply:
Maybe I'm misunderstanding but I'm trying to emulate the current code pattern where for example there are separate types for `Post` and `Comment` (though they have `AuthoredContent` type for common fields) and separate `searchResultToPost` and `searchResultToComment` functions to convert the ES search results to their respective types.

Reply:
to get around flow complaining fields can be typed as optional. yeah, I'm not sure exactly how we do it, I'm just seeing how much code is being added just to add one more type of result (bootcamps) and that has me worried about dealing with tons of similar code when we add programs and learning paths as well, so I think if we can get out in front of it by trying to re-use as much as possible we'll be in a better position going forward.

it seems to me that the display we want to have in the learning resource card is mostly the same for all of these types of resources, and then there are differences in how we want to display them in the learning resource drawer. to me the way to handle that that makes things easiest is to treat these things the same way everywhere except where we actually need to render something different, like the list of things in the drawer display for a learning path or similar. it's a bit analogous to how we don't worry about whether a post is a link post or a text post unless we're actually at the point of rendering it's content.

anyhow, if you don't agree that's fine, we can keep this as-is for now. I just want to make sure we don't end up with a lot of very similar code for handling these different things that's hard to maintain later.

Reply:
I think I might understand now, the `LearningResource` type would be independent of the existing `Bootcamp` and `Course` types (still needed for the expanded display in the drawer) and have only the fields displayed in the compact card?

Reply:
I did the above